crates/pathfinder/src/rpc.rs

```diff
@@ -366,6 +362,34 @@ Hint: If you are looking to run two instances of pathfinder, you must configure
         let module = module.into_inner();
         Ok(server.start(module).map(|handle| (handle, local_addr))?)
     }
+
+    fn register_method<T: RpcMethod>(
+        module: &mut RpcModuleWrapper<RpcApi>,
+    ) -> Result<
+        jsonrpsee::core::server::rpc_module::MethodResourcesBuilder<'_>,
+        jsonrpsee::core::Error,
+    > {
+        module.register_async_method(T::NAME, |params, context| async move {
+            let input = params.parse::<T::Input>()?;
+            T::execute(context, input).await.map_err(|e| {
+                let rpc_err: error::RpcError = e.into();
+                jsonrpsee::core::Error::from(rpc_err)
+            })
+        })
+    }
+}
```

Comment:
I haven't compiled it though :sweat_smile: 
```suggestion
    fn register_method<T: RpcMethod>(
        module: &mut RpcModuleWrapper<RpcApi>,
    ) -> Result<
        jsonrpsee::core::server::rpc_module::MethodResourcesBuilder<'_>,
        anyhow::Error,
    > {
        module.register_async_method(T::NAME, |params, context| async move {
            let input = params.parse::<T::Input>()?;
            T::execute(context, input).await.map_err(|e| {
                let rpc_err: error::RpcError = e.into();
                jsonrpsee::core::Error::from(rpc_err)
            })
        })
        .with_context(|| format!("Registering {}", T::NAME))
    }
}
```